generators/server/prompts.js

```diff
@@ -101,6 +101,31 @@ function askForServerSideOpts() {
         },
         {
             when: function (response) {
+                return applicationType === 'gateway' || applicationType === 'microservice' || applicationType === 'uaa';
+            },
+            type: 'list',
+            name: 'serviceDiscoveryType',
+            message: function (response) {
+                return getNumberedQuestion('Which Service Discovery and Configuration solution would you like to use?', applicationType === 'gateway' || applicationType === 'microservice');
+            },
+            choices: [
+                {
+                    value: 'eureka',
+                    name: 'JHipster Registry (using Eureka and Spring Cloud Config)'
+                },
+                {
+                    value: 'consul',
+                    name: 'Consul (using Spring Cloud Consul)'
+                },
+                {
+                    value: 'no',
```

Comment:
can we use false instead of no please so that its aligned with other options we have